File: classes/parte-7-paradigmas/111-herencia-composicion-y-polimorfismo/implementaciones/python/main.py
import sys


class Perro:
    def sonido(self):
        return "guau"


class Gato:
    def sonido(self):
        return "miau"


class Vaca:
    def sonido(self):
        return "muu"


tipo = sys.stdin.readline().strip()
animales = {"perro": Perro(), "gato": Gato(), "vaca": Vaca()}
print(f"sonido={animales[tipo].sonido()}")

File: classes/parte-7-paradigmas/111-herencia-composicion-y-polimorfismo/implementaciones/php/main.php
<?php
interface Animal { public function sonido(): string; }
class Perro implements Animal { public function sonido(): string { return "guau"; } }
class Gato implements Animal { public function sonido(): string { return "miau"; } }
class Vaca implements Animal { public function sonido(): string { return "muu"; } }

$tipo = trim(fgets(STDIN));
$animales = ["perro" => new Perro(), "gato" => new Gato(), "vaca" => new Vaca()];
echo "sonido=" . $animales[$tipo]->sonido() . "\n";
